src/AutoRest.CSharp/LowLevel/Generation/LowLevelModelWriter.cs

```diff
@@ -48,9 +49,19 @@ private static void WritePublicConstructor(CodeWriter writer, ConstructorSignatu
             {
                 var initializedFields = WriteFieldsInitialization(writer, signature, model);
                 // TODO: Add IReadOnlyDictionary
-                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f) && TypeFactory.IsReadOnlyList(f.Type)))
+                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f)))
                 {
-                    writer.Line($"{field.Name:I} = new List<{field.Type.Arguments[0]}>(0).AsReadOnly();");
+                    if (TypeFactory.IsList(field.Type))
+                    {
+                        if (TypeFactory.IsReadOnlyList(field.Type))
+                        {
+                            writer.Line($"{field.Name:I} = new List<{field.Type.Arguments[0]}>(0).AsReadOnly();");
```

Comment:
Why can't we use `Array.Empty<field.Type.Arguments[0]>()` here?

Reply:
I guess it's due to the performance: https://www.tabsoverspaces.com/233833-best-way-to-create-an-empty-collection-array-and-list-in-csharp-net

Reply:
According to this article, `Array.Empty` is the best (which is obvious cause it is a static field). To be clear, the output line should look like this: `OptionalStringList = Array.Empty<string>();`